Dockerfile.critical_css

```diff
@@ -0,0 +1,47 @@
+ARG image_tag=latest
+ARG node_version
+FROM pinterb/jq:0.0.16 AS jq
+FROM elifesciences/journal_npm:${image_tag} as npm
+
+FROM node:${node_version}
+
+# From list at https://developers.google.com/web/tools/puppeteer/troubleshooting#chrome_headless_doesnt_launch
+RUN apt-get update && apt-get install --no-install-recommends -y \
+    fonts-liberation \
+    gconf-service \
+    libappindicator1 \
+    libasound2 \
+    libatk1.0-0 \
+    libcups2 \
+    libdbus-1-3 \
+    libgconf-2-4 \
+    libgtk-3-0 \
+    libnspr4 \
+    libnss3 \
+    libx11-xcb1 \
+    libxcomposite1 \
+    libxcursor1 \
+    libxdamage1 \
+    libxfixes3 \
+    libxi6 \
+    libxrandr2 \
+    libxss1 \
+    libxtst6 \
+    locales \
+    lsb-release \
+    unzip \
+    xdg-utils \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=jq /usr/local/bin/jq /usr/bin/jq
+
+RUN mkdir -p build/critical-css
+
+COPY --from=npm /node_modules/ node_modules/
+
+COPY check_critical_css.sh \
+    critical-css.json \
+    gulpfile.js \
+    ./
+
+CMD node_modules/.bin/gulp critical-css:generate && ./check_critical_css.sh
```

Comment:
The `CMD` is useful for `docker-compose run` executions, but unless it is overridden with `command:` every `docker-compose up` regenerates the critical-css, which makes startup slow, and can happen during other smoke tests rather than explicitly as a pipeline stage. Without separating the service into yet another `docker-compose.critical.yml`, it could suffice to put an empty default here.

Reply:
All use cases I have tried work well so this is the last outstanding issue

Reply:
Seems that it was in `docker-compose.override.yml` so would still execute it on `up` :thinking: No one reads READMEs so I'd default to an explicit `run node_modules/...`, wrapped in a script inside or outside the container if it's too lengthy. We can imagine casual users wondering whether that container should be running, or why it stopped; I would expect not to use critical CSS locally in the normal workflow.

Reply:
We’d really want https://github.com/docker/compose/issues/1896 to avoid automatically starting, but that’s been rejected. Only possibility without fiddling with the `command` is a separate `docker-compose.critical-css.yml` which adds more complexity.

I don’t think there’s that much cost to leaving it as it. It won’t slow anything down. If it does become an issue we can revisit.